src/proc_interrupts.c

```diff
@@ -124,21 +143,20 @@ int do_proc_interrupts(int update_every, unsigned long long dt) {
     // --------------------------------------------------------------------
 
     st = rrdset_find_bytype("system", "interrupts");
-    if(unlikely(!st)) {
-        st = rrdset_create("system", "interrupts", NULL, "interrupts", NULL, "System interrupts", "interrupts/s", 1000, update_every, RRDSET_TYPE_STACKED);
-
-        for(l = 0; l < lines ;l++) {
-            struct interrupt *irr = irrindex(irrs, l, cpus);
-            if(unlikely(!irr->used)) continue;
-            rrddim_add(st, irr->id, irr->name, 1, 1, RRDDIM_INCREMENTAL);
-        }
-    }
+    if(unlikely(!st)) st = rrdset_create("system", "interrupts", NULL, "interrupts", NULL, "System interrupts", "interrupts/s", 1000, update_every, RRDSET_TYPE_STACKED);
     else rrdset_next(st);
 
     for(l = 0; l < lines ;l++) {
         struct interrupt *irr = irrindex(irrs, l, cpus);
         if(unlikely(!irr->used)) continue;
-        rrddim_set(st, irr->id, irr->total);
+        if(unlikely(!irr->rd || strncmp(irr->rd->name, irr->name, MAX_INTERRUPT_NAME) != 0)) {
```

Comment:
As I said before, any line can be shifted by an interrupt insertion or removal, making our `RRDDIM` pointers incorrect (they then belong to another line). This is why a 10s check is incorrect, we would add incorrect values for a few seconds in the wrong `RRDDIM`.

I think those `strncmp()` would average to about 3 or 4 `char` comparisons by line, we might consider it acceptable given the more heavy parsing and copies of string done before in the function. If you have a better idea, I'm all ears (I'm not sure we would gain much by using a name CRC for example)